server/fishtest/rundb.py

```diff
@@ -427,10 +427,9 @@ def scavenge(self, run):
         # print("scavenge ", run["_id"])
         dead_task = False
         old = datetime.utcnow() - timedelta(minutes=6)
-        task_id = -1
         run_id = str(run["_id"])
-        for task in run["tasks"]:
-            task_id += 1
+        for task_id in range(0, len(run["tasks"])):
```

Comment:
```python
    old = datetime.utcnow() - timedelta(minutes=6)
    for task_id, task in enumerate(run["tasks"]):
        if task["active"] and task["last_updated"] < old:
            task["active"] = False
            dead_task = True
```

Reply:
Is `enumerate` thread safe when other threads add elements to the enumerated without locking?

Reply:
The run is locked, I'm expecting that nobody can add a task in that run.

Reply:
The run is locked how?